src/tracking_results.jl

```diff
@@ -11,6 +11,7 @@ struct TrackingResults{
     }
     state::TS
     correlator::C
+    filtered_prompt::ComplexF64
```

Comment:
Is there a special reason for putting the filtered prompt signal in the tracking results? In my opinion, this is quite unsafe to evaluate, as the tracking loop might generate more than one prompt (or even no).

Reply:
Yes, it's like `correlator`, but filtered. `correlator` holds the last valid correlation results (after a complete code length has been integrated). The same applies to filtered prompt. `filtered_prompt` is needed because in some circumstances the filter can not be reconstructed outside of track. The implementation here allows to update the states of the post correlation filter every time there are new correlation results. Therefore, the state of the post correlation filter might be different at the time that `track` returns.